src/compiler/legend.ts

```diff
@@ -45,7 +44,7 @@ export function compileLegend(model: Model, channel: Channel, def) {
   });
 
   // 2) Add mark property definition groups
-  const props =  isLegend(legend) && legend.properties || {};
+  const props = (typeof legend !== 'boolean' && legend.properties) || {};
```

Comment:
`true instanceof Boolean`  is `false`

`instanceof` is for checking prototype chain.  `typeof` is the right way to test primitive type.